src/host/tcp.h

```diff
@@ -589,7 +601,13 @@ namespace asynchost
 
     static void on_resolved(uv_getaddrinfo_t* req, int rc, struct addrinfo*)
     {
-      static_cast<TCPImpl*>(req->data)->on_resolved(req, rc);
+      std::unique_lock<std::mutex> guard(pending_resolve_requests_mtx);
+      pending_resolve_requests.erase(req);
```

Comment:
Should the lock be released after this line?

Reply:
No! See previous (resolved) comment by Eddy. In theory (but not in the current usage pattern) it may become possible that `~TCPImpl` runs between check and use of `req->data`.